src/main/java/mate/academy/web/filter/AuthorizationFilter.java

```diff
@@ -0,0 +1,78 @@
+package mate.academy.web.filter;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.academy.controllers.LoginController;
+import mate.academy.lib.Injector;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+
+public class AuthorizationFilter implements Filter {
+    private static final Injector injector = Injector.getInstance("mate.academy");
+    private static final UserService userService = (UserService) injector
+            .getInstance(UserService.class);
+    private Map<String, List<Role.RoleName>> protectedUrls = new HashMap<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/users/all", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/users/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/add", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/all-admin", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/all-admin", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/shopping-carts/show", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-carts/products/add", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-carts/products/delete", List.of(Role.RoleName.USER));
+        protectedUrls.put("/orders/complete", List.of(Role.RoleName.USER));
+        protectedUrls.put("/orders/all-user", List.of(Role.RoleName.USER));
```

Comment:
```suggestion
        protectedUrls.put("user/orders", List.of(Role.RoleName.USER));
```
if you want to mark in url which urls accessible  to user and which are for admin, then let's put "admin" in those which are for admin(if you have the same url for user)

Reply:
done)